components/server/ee/src/user/stripe-service.ts

```diff
@@ -149,11 +180,27 @@ export class StripeService {
         }
         const startOfNextMonth = new Date(new Date().toISOString().slice(0, 7) + "-01"); // First day of this month (YYYY-MM-01)
         startOfNextMonth.setMonth(startOfNextMonth.getMonth() + 1); // Add one month
-        await this.getStripe().subscriptions.create({
-            customer: customer.id,
-            items: [{ price: priceId }],
-            automatic_tax: { enabled: isAutomaticTaxSupported },
-            billing_cycle_anchor: Math.round(startOfNextMonth.getTime() / 1000),
+
+        await reportStripeOutcome("subscriptions_create", () => {
+            return this.getStripe().subscriptions.create({
+                customer: customer.id,
+                items: [{ price: priceId }],
+                automatic_tax: { enabled: isAutomaticTaxSupported },
+                billing_cycle_anchor: Math.round(startOfNextMonth.getTime() / 1000),
+            });
         });
     }
 }
+
+async function reportStripeOutcome<T>(op: string, f: () => Promise<T>) {
+    const timer = stripeClientRequestsCompletedDurationSeconds.startTimer();
+    let outcome = "ok";
+    try {
+        return await f();
+    } catch (e) {
+        outcome = e.type;
```

Comment:
e has `any` type here. How do we know that it has a `type` field?

Reply:
We don't. But you're right, outcome should be `string | undefined`, and ` observeStripeClientRequestsCompleted` might need minimal extension to translate `undefined` to `"undefined"` (or whatever we want to call it).

Reply:
Interesting, I'd have hoped TS has my back here but looks like no. 

I mostly followed https://stripe.com/docs/api/errors/handling?lang=node

Will add a guard

Reply:
Added.